src/Microsoft.OpenApi.Readers/OpenApiStreamReader.cs

```diff
@@ -60,6 +61,21 @@ public OpenApiDocument Read(Stream input, out OpenApiDiagnostic diagnostic)
             // Parse the OpenAPI Document
             var document = context.Parse(yamlDocument, diagnostic);
 
+            // Resolve References if requested
+            switch (_settings.ReferenceResolution) 
+            {
+                case ReferenceResolutionSetting.ResolveRemoteReferences:
+                    throw new ArgumentException(Properties.SRResource.CannotResolveRemoteReferencesSynchronously);
```

Comment:
>throw new ArgumentException(Properties.SRResource.CannotResolveRemoteReferencesSynchronously); [](start = 20, length = 94)

This will be supported, right? Can we just go with NotImplementedException()?

Reply:
I don't think this will be handled.  If you want to resolve remote references then you should call the ReadAsync method instead.  The ReadAsync will also load related documents into a workspace.

Reply:
Makes sense. 

---
In reply to: [167757446](https://github.com/Microsoft/OpenAPI.NET/pull/185#discussion_r167757446) [](ancestors = 167757446)